### PR TITLE
Add eslint to the backend

### LINKED ISSUE #73: Add eslint for backend

### PR DESCRIPTION
**Pull request by AbhiPrasad**
Closes #73 

Adds eslint to the backend. We can now `npm run lint` on the backend! 

Travis CI coming soon, I've request permission from the uwblueprint org owners to allow it access to our repos. 

Sorry again for the lack of stuff this weekend, got caught up with a lot. Will hopefully try to push a lot this week.

### CODE REVIEW

File: backend/app.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
 const express = require('express');
-const app = express();
+var mongoose = require('mongoose');
```
Comment (theRoughCode) [inline backend/app.js]: Would a `const` here work?

File: backend/controllers/faq.js [hunk 2]
```diff
@@ -41,10 +43,14 @@ faqRouter.post('/insert', function(req, res) {
...
-		res.json({
-			message: 'Question created!',
-	 	   faq: faq
-	   });
+		if (err) {
+			res.json(err);
+		} else {
+			res.json({
+				message: 'Question created!',
+				faq: faq
```
Comment (theRoughCode) [inline backend/controllers/faq.js]: Can use Es6 syntax: `faq` without specifying pair.

File: backend/package.json [hunk 3]
```diff
@@ -18,11 +19,18 @@
   },
   "homepage": "https://github.com/uwblueprint/water-festival#readme",
   "dependencies": {
+    "bluebird": "^3.5.1",
```
Comment (theRoughCode) [inline backend/package.json]: 👍

File: backend/package.json [hunk 4]
```diff
@@ -18,11 +19,18 @@
...
     "body-parser": "^1.18.2",
     "cors": "^2.8.4",
     "dotenv": "^5.0.0",
     "express": "^4.16.1",
     "fuse.js": "^3.1.0",
+    "mongodb": "^3.0.2",
```
Comment (theRoughCode) [inline backend/package.json]: Do we need this dependency?

Comment (AbhiPrasad) [reply backend/package.json]: Yeah, we use it here https://github.com/uwblueprint/water-festival/blob/12c252f3802233846d146e58398818cdb52b7745/backend/controllers/Events.js#L4

File: backend/app.js (hunk 1 above)
Comment (AbhiPrasad) [reply backend/app.js]: Yeah, there's a bunch of instances throughout the codebase for that. I'll open up an issue for it.

Comment (AbhiPrasad) [reply backend/app.js]: On second thought, I'll fix it here. We can probably just bring it up in code reviews.

File: backend/controllers/Events.js [hunk 5]
```diff
@@ -52,6 +52,10 @@ eventRouter.post('/insert', function(req, res) {
   event.imageURI = req.body.imageURI;
 
   event.save(function(err) {
+    if (err) {
+      res.status(500).json(err);
```
Comment (amyqiu) [inline backend/controllers/Events.js]: Good catch!

**Pull request merged**